src/jaxsim/mujoco/utils.py

```diff
@@ -0,0 +1,105 @@
+        mujoco_data: An optional Mujoco data object. If None, a new one will be created.
+        update_removed_joints:
+            If True, the positions of the joints that have been removed during the
+            model reduction process will be set to their initial values.
+
+    Returns:
+        The Mujoco data object containing the state of the JaxSim model.
+
+    Note:
+        This method is useful to initialize a Mujoco data object used for visualization
+        with the state of a JaxSim model. In particular, this function takes care of
+        initializing the positions of the joints that have been removed during the
+        model reduction process. After the initial creation of the Mujoco data object,
+        it's faster to update the state using an external MujocoModelHelper object.
+    """
+
+    # The package `jaxsim.mujoco` is supposed to be jax-independent.
+    # We import all the JaxSim resources privately.
+    import jaxsim.api as js
+
+    if not isinstance(jaxsim_model, js.model.JaxSimModel):
+        raise ValueError("The `jaxsim_model` argument must be a JaxSimModel object.")
+
+    if not isinstance(jaxsim_data, js.data.JaxSimModelData):
+        raise ValueError("The `jaxsim_data` argument must be a JaxSimModelData object.")
+
+    # Create the helper to operate on the Mujoco model and data.
+    model_helper = MujocoModelHelper(model=mujoco_model, data=mujoco_data)
+
+    # Set the model position.
+    if jaxsim_model.floating_base():
+
+        model_helper.set_base_position(position=np.array(jaxsim_data.base_position()))
+
+    # Set the model orientation.
+    if jaxsim_model.floating_base():
+
+        model_helper.set_base_orientation(
+            orientation=np.array(jaxsim_data.base_orientation())
+        )
+
+    # Set the joint positions.
+    if jaxsim_model.dofs() > 0:
+
+        model_helper.set_joint_positions(
+            joint_names=list(jaxsim_model.joint_names()),
+            positions=np.array(
+                jaxsim_data.joint_positions(
+                    model=jaxsim_model, joint_names=jaxsim_model.joint_names()
+                )
+            ),
+        )
+
+    # Updating these joints is not necessary after the first time.
+    # Users can disable this update after initialization.
+    if update_removed_joints:
+
+        # Create a dictionary with the joints that have been removed for various reasons
+        # (like link lumping due to model reduction).
+        joints_removed_dict = {
+            j.name: j
+            for j in jaxsim_model.description._joints_removed
+            if j.name not in set(jaxsim_model.joint_names())
+        }
+
+        # Get all the joints in the Mujoco and JaxSim models.
+        joints_mujoco = set(model_helper.joint_names())
+
+        # Set the positions of the removed joints.
+        for joint_name in set(joints_removed_dict.keys()):
+
+            if joint_name not in joints_mujoco:
+                continue
```

Comment:
Is there a possibility that there are duplicates in `joints_removed_dict.keys()`? In that case we could either do:
```suggestion
        for joint_name in joints_removed_dict:

            if joint_name not in joints_mujoco:
                continue
```

or just take the intersection:

```suggestion
        for joint_name in set(joints_removed_dict.keys()) & joints_mujoco:
```

Reply:
Mmh if the `set` was there, maybe there was a case in which it was needed. Now I do not remember, this helper is part of my local repo for a few months now :) I removed the set for the time being, we can add it back if needed.